### PR TITLE
Feature/#646 bis

### PR DESCRIPTION
**Pull request by monsieurtanuki**

### CODE REVIEW

**Comment by monsieurtanuki** [comment]
@secaw Sorry for the very late #646 answer: I got drowned into #329 and #725.
I added your main suggestions - the rest seemed a bit too specific.

**Comment by secaw** [comment]
Well, glad, I could contribute just a tiny bit :-). Great more work by the way, thanks!

File: osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTileProviderBasic.java [hunk 1]
```diff
@@ -127,6 +127,7 @@ public void detach(){
 	 */
 	@Override
 	protected boolean isDowngradedMode() {
-		return mNetworkAvailabilityCheck != null && !mNetworkAvailabilityCheck.getNetworkAvailable();
+		return (mNetworkAvailabilityCheck != null && !mNetworkAvailabilityCheck.getNetworkAvailable())
```
Comment (spyhunter99) [inline osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTileProviderBasic.java]: is this change somehow related to #646?

Comment (monsieurtanuki) [reply osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTileProviderBasic.java]: Indeed: we discussed that @secaw and I, and you can find more details at the end of #646.
